azurerm/resource_arm_relay_hybrid_connection_test.go

```diff
@@ -0,0 +1,258 @@
+					testCheckAzureRMHybridConnectionExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "requires_client_authorization"),
+					resource.TestCheckResourceAttr(resourceName, "user_metadata", "metadatatest"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMHybridConnection_update(t *testing.T) {
+	resourceName := "azurerm_relay_hybrid_connection.test"
+	rInt := tf.AccRandTimeInt()
+	location := testLocation()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMHybridConnectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMHybridConnection_basic(rInt, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHybridConnectionExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "requires_client_authorization"),
+				),
+			},
+			{
+				Config: testAccAzureRMHybridConnection_update(rInt, location),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "requires_client_authorization", "false"),
+					resource.TestCheckResourceAttr(resourceName, "user_metadata", "metadataupdated"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMHybridConnection_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	resourceName := "azurerm_relay_hybrid_connection.test"
+	ri := tf.AccRandTimeInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMHybridConnectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMHybridConnection_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHybridConnectionExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "requires_client_authorization"),
+				),
+			},
+			{
+				Config:      testAccAzureRMHybridConnection_requiresImport(ri, testLocation()),
+				ExpectError: testRequiresImportError("azurerm_relay_hybrid_connection"),
+			},
+		},
+	})
+}
+
+func testAccAzureRMHybridConnection_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_relay_namespace" "test" {
+  name                = "acctestrn-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  sku_name = "Standard"
+}
+
+resource "azurerm_relay_hybrid_connection" "test" {
+	name                 = "acctestrnhc-%d"
+	resource_group_name = "${azurerm_resource_group.test.name}"
```

Comment:
Adding a space so the equals sign lines up a little better
```suggestion
	resource_group_name  = "${azurerm_resource_group.test.name}"
```